compiler/noirc_errors/src/debug_info.rs

```diff
@@ -42,4 +48,39 @@ impl DebugInfo {
     pub fn opcode_location(&self, loc: &OpcodeLocation) -> Option<Vec<Location>> {
         self.locations.get(loc).cloned()
     }
+
+    pub fn count_span_opcodes(&self) -> HashMap<&Location, OpCodesCount> {
+        let mut accumulator: HashMap<&Location, Vec<&OpcodeLocation>> = HashMap::new();
+
+        for (opcode_location, locations) in self.locations.iter() {
```

Comment:
may be you want to use only the first location?

Reply:
I am closing this, it works as intended as it accumulate the locations 'calling' the opcode (from the call stack)